### PR TITLE
[DEV-3984] Remove references_location

### PR DESCRIPTION
**Pull request by kbard**
**Description:**

`references_location` is an obsoleteish table that we've been wanting to eradicate for quite some time now.  This is the pull request that attempts to accomplish that.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation should be unaffected
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [x] Matview impact assessment completed
5. [x] Frontend should be unaffected
6. [x] Data validation completed
7. [x] No Operations ticket required
8. [x] Jira Ticket [DEV-3984](https://federal-spending-transparency.atlassian.net/browse/DEV-3984):
    - [x] Link to this Pull-Request

### CODE REVIEW

File: usaspending_api/database_scripts/matview_generator/summary_state_view.json [hunk 1]
```diff
@@ -27,19 +27,18 @@
     "LEFT OUTER JOIN",
     "  transaction_fpds ON (transaction_normalized.id = transaction_fpds.transaction_id)",
     "LEFT OUTER JOIN",
-    "  references_location AS place_of_performance ON (transaction_normalized.place_of_performance_id = place_of_performance.location_id)",
+    "  transaction_fabs ON (transaction_normalized.id = transaction_fabs.transaction_id)",
     "WHERE",
     "  transaction_normalized.action_date >= '2007-10-01' AND",
-    "  place_of_performance.location_country_code = 'USA' AND",
-    "  place_of_performance.state_code IS NOT NULL",
+    "  COALESCE(transaction_fpds.place_of_perform_country_c, transaction_fabs.place_of_perform_country_c, 'USA') = 'USA' AND",
```
Comment (BrianZito) [inline usaspending_api/database_scripts/matview_generator/summary_state_view.json]: Previous logic did not default to USA

File: usaspending_api/etl/transaction_loaders/derived_field_functions_fpds.py [hunk 2]
```diff
@@ -3,12 +3,11 @@
...
 from usaspending_api.common.helpers.date_helper import fy
 from usaspending_api.etl.transaction_loaders.cached_reference_data import subtier_agency_list
-from usaspending_api.etl.transaction_loaders.data_load_helpers import capitalize_if_string
 from usaspending_api.broker.helpers.get_business_categories import get_business_categories
 from usaspending_api.common.helpers.date_helper import cast_datetime_to_utc
-from usaspending_api.references.abbreviations import code_to_state, state_to_code
 
-ZIP_CODE_PATTERN = re.compile(r"^(\d{5})\-?(\d{4})?$")
+
+ZIP_CODE_PATTERN = re.compile(r"^(\d{5})-?(\d{4})?$")
```
Comment (BrianZito) [inline usaspending_api/etl/transaction_loaders/derived_field_functions_fpds.py]: It doesn't look like we use this variable anymore.

File: usaspending_api/etl/transaction_loaders/tests/test_fpds_loader.py [hunk 3]
```diff
@@ -209,60 +201,33 @@ def test_load_from_broker():
 # This only runs for one of the most simple tables, since this is supposed to be a test to ensure that the loader works,
 # NOT that the map is accurate
 def test_create_load_object(monkeypatch):
-    custom_bools = {"is_awesome": "awesome", "is_not_awesome": "nawesome", "reasons_to_not_be_awesome": "noawersn"}
+    columns = {"input_field_1": "output_field_1", "input_field_2": "output_field_2"}
+    bools = {"input_field_3": "output_field_3", "input_field_4": "output_field_4", "input_field_5": "output_field_5"}
+    functions = {"input_field_6": lambda t: t["input_field_6"] * 2, "input_field_7": lambda t: "replacement"}
```
Comment (BrianZito) [inline usaspending_api/etl/transaction_loaders/tests/test_fpds_loader.py]:
```suggestion
    functions = {"output_field_6": lambda t: t["input_field_6"] * 2, "output_field_7": lambda t: "replacement"}
```
Functions are backwards relative to the other maps, probably a mistake.

File: usaspending_api/database_scripts/matview_generator/summary_state_view.json (hunk 1 above)
Comment (kbard) [reply usaspending_api/database_scripts/matview_generator/summary_state_view.json]: I believe the previous logic was faulty.  Pretty much everywhere else missing country codes are treated as USA.  We had a ticket many moons ago that stated that as a requirement, but this particular matview was not listed.

https://federal-spending-transparency.atlassian.net/browse/DEV-2911

Comment (BrianZito) [reply usaspending_api/database_scripts/matview_generator/summary_state_view.json]: Alright, that checks out. I'd still advise a technical note at the top like for the legal_entity PR

**Comment by BrianZito** [comment]
I'll hold back from merging in case we want more eyes on the PR.

**Comment by kbard** [comment]
> This branch has conflicts that must be resolved

:shakes fist:

**Pull request merged**